packages/core-contracts/src/contracts/arks/MidasArk.sol

```diff
@@ -0,0 +1,342 @@
+     * @dev 50 is 50% (50/10000)
+     */
+    uint256 public constant DEFAULT_SLIPPAGE = 50;
+
+    /**
+     * @notice Rounding offset for Midas calculations
+     */
+    uint256 public constant MIDAS_ROUNDING_OFFSET = 1;
+
+    /*//////////////////////////////////////////////////////////////
+                           IMMUTABLE STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    IDepositVault public immutable issuanceVault;
+    IMToken public immutable mToken;
+    IRedemptionVault public immutable redemptionVault;
+    IMidasOracle public immutable oracle;
+
+    /**
+     * @notice The conversion factor from the underlying asset to the mToken
+     * @dev e.g. 1e12 is the conversion factor for USDC (mTokens have 18 decimals)
+     */
+    uint256 public immutable TO_M_TOKEN_DECIMALS;
+
+    /*//////////////////////////////////////////////////////////////
+                           MUTABLE STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The request ID for the redemption
+    uint256 public withdrawalRequestId;
+
+    /*//////////////////////////////////////////////////////////////
+                           CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Constructor to set up the MidasArk
+     * @param _mToken Address of the Midas mToken (for deposits and withdrawals)
+     * @param _issuanceVault Address of the Midas Issuance Vault (for deposits)
+     * @param _redemptionVault Address of the Midas Redemption Vault (for withdrawals)
+     * @param _params ArkParams struct containing necessary parameters for Ark initialization
+     */
+    constructor(
+        address _mToken,
+        address _issuanceVault,
+        address _redemptionVault,
+        ArkParams memory _params
+    ) ArkWithWithdrawalRequest(_params, DEFAULT_SLIPPAGE) {
+        if (_mToken == address(0)) revert InvalidVaultAddress();
+        if (_issuanceVault == address(0)) revert InvalidIssuanceVault();
+        if (_redemptionVault == address(0)) revert InvalidRedemptionVault();
+
+        issuanceVault = IDepositVault(_issuanceVault);
+        redemptionVault = IRedemptionVault(_redemptionVault);
+        mToken = IMToken(_mToken);
+        oracle = IMidasOracle(issuanceVault.mTokenDataFeed());
+
+        // Verify mToken matches vaults
+        IMToken issuanceVaultMToken = issuanceVault.mToken();
+        if (address(issuanceVaultMToken) != _mToken)
+            revert InvalidMTokenAddress();
+
+        IMToken redemptionVaultMToken = redemptionVault.mToken();
+        if (address(redemptionVaultMToken) != _mToken)
+            revert InvalidMTokenAddress();
+
+        // Calculate conversion factor
+        uint8 mTokenDecimals = mToken.decimals();
+        uint8 assetDecimals = IERC20Metadata(address(config.asset)).decimals();
+        if (mTokenDecimals < assetDecimals) revert InvalidMTokenDecimals();
```

Comment:
Why is this? We could have the converting factor accepting mToken having less decimals than the asset.